Add secrets scanning documentation (#27892)
DataDog/documentation: content/en/security/code_security/secret_scanning/github_actions.md

---
description: Use Datadog and GitHub to detect Secrets in code in a CI pipeline.
is_beta: true
title: Secrets Scanning and GitHub Actions
---
## Overview

> Run a [Datadog Secrets Scanning][1] job in your GitHub Action workflows. This action wraps the [Datadog Static Analyzer][8],
> that scans for secrets, invokes it against your codebase, and uploads the results to Datadog.

Reviewer: ```suggestion
Run a [Datadog secrets scanning][1] job in your GitHub Action workflows. This action wraps the [Datadog Static Analyzer][8] (that scans for secrets), invokes it against your codebase, and uploads the results to Datadog.
```

## Workflow

> Create a file in `.github/workflows` to run a Datadog Static Code Analysis job.

Reviewer: we call it a Datadog secrets scanning job and now a Datadog Static Code Analysis job. I think the former is clearer to the purpose of the topic. Let me know if that's a mistake.

Author: You updated and that seems totally fine for me! Thanks for checking and fixing it!

> Create a file in `.github/workflows` to run a Datadog Static Code Analysis job.

Reviewer: ```suggestion
Create a file in `.github/workflows` to run a Datadog secrets scanning job.
```

The following is a sample workflow file.

```yaml
on: [push]

jobs:
  check-quality:
    runs-on: ubuntu-latest
    name: Datadog Static Analyzer
    steps:
      - name: Checkout
        uses: actions/checkout@v4
      - name: Check code meets quality standards
        id: datadog-static-analysis
        uses: DataDog/datadog-static-analyzer-github-action@v1
        with:
          dd_app_key: ${{ secrets.DD_APP_KEY }}
          dd_api_key: ${{ secrets.DD_API_KEY }}
          dd_site: "datadoghq.com"
          cpu_count: 2
          enable_performance_statistics: false
          static_analysis_enabled: false
          secrets_enabled: true
```

> You **must** set your Datadog API and application keys as [secrets in your GitHub repository][4] whether at the organization or repository level. Ensure that you add the `code_analysis_read` scope to your Datadog application key. For more information, see [API and Application Keys][2].

Reviewer: ```suggestion
You **must** set your Datadog API and application keys as [secrets in your GitHub repository][4], at either the organization or repository level. Ensure that you add the `code_analysis_read` scope to your Datadog application key. For more information, see [API and Application Keys][2].
```

> Make sure to replace `dd_site` with the Datadog site you are using[3].

Reviewer: ```suggestion
Make sure to replace `dd_site` with the Datadog site you are using.
```
